templates/taxi/driver_list.html

```diff
@@ -5,10 +5,17 @@ <h1>Driver list</h1>
   {% if driver_list %}
     <ul>
       {% for driver in driver_list%}
-        <li>
-          <a href="{% url "taxi:driver-detail" pk=driver.id %}">{{ driver.username }}</a>
+        {% if driver.username == user.username %}
```

Comment:
as far as I remember, driver and user are the same models, thus you can just compare driver == user and django will check if it's the same objects